# Coefficient of determination:
* The coefficient of determination is R^2 or r-squared.
* It is the statistical measure in a regression model.
* It helps in determining the proportion of variance in the dependent variable that is explained by the independent variable.
* It tells how well the data fits the model in other words, it tells the goodness of fit.

# Importing libraries

In [7]:
from sklearn import datasets
from sklearn.model_selection import train_test_split
import numpy as np
from sklearn.metrics import r2_score
from sklearn.linear_model import LogisticRegression

Loading the iris dataset

In [2]:
iris = datasets.load_iris()

Splitting the dataset to train and test data.

In [10]:
X_train, X_test, y_train, y_test = train_test_split(iris.data, iris.target, random_state = 42, test_size = 0.2)

Training the data with logistic regression

In [8]:
model=LogisticRegression()

In [12]:
model.fit(X_train,y_train)

LogisticRegression()

model evaluation

In [15]:
x_train_prediction =model.predict(X_train)
r2 = r2_score(x_train_prediction, y_train)
print('r2 score for perfect model is', (r2)*100)

r2 score for perfect model is 96.33997559983733


# Coclusion:

Here, the value of r2 is 96% and the value is > 85%.
We can conclude by saying, logistic regression is the good fit for the model.